spec/controllers/posts_controller_spec.rb

```diff
@@ -0,0 +1,35 @@
+require 'rails_helper'
+
+RSpec.describe 'PostsController', type: :request do
+  describe 'post controller' do
+    it 'redirects to the Post show page ' do
+      get '/users/:user_id/posts/:id'
+      expect(response).to render_template(:show)
+    end
+
+    it "redirects to the Post 'index' page" do
+      get '/users/:user_id/posts'
+      expect(response).to render_template(:index)
+    end
+
+    it 'confirms status code ok for post show page' do
+      get '/users/:user_id/posts/:id'
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'show the exact text on post show page' do
+      get '/users/:user_id/posts/:id'
+      expect(response.body).to include('Show method for posts')
+    end
+
+    it 'show the exact text on post index page' do
+      get '/users/:user_id/posts'
+      expect(response.body).to include('Here is a list of posts for a given user')
+    end
+
+    it 'does not render a different template' do
+      get '/users/:user_id/posts/:id'
+      expect(response).to_not render_template(:index)
+    end
+  end
+end
```

Comment:
- [ ] Kindly double check the posts_controller because all the tests here are failing(Note: Only tests on posts_controller are failing). Kindly check if you have any uncommitted code on your local computer  ![image](https://user-images.githubusercontent.com/62705947/214434739-2e2c5bc4-a679-4153-b9f8-013f06a002a8.png)